### ISSUE TITLE
running warpx with python inputs

### ISSUE THREAD

**Issue by rl3418** [ECP-WarpX/WarpX#4688]
I wish to know how to run WarpX through a python input. WarpX is built as a static library. Python import searches for .so file. It doesn't recognise the .a files for static libraries. Is there an option to built WarpX as a dynamic library? Or is there an easy way to import static libraries?

**Comment by rl3418** [comment]
I looked through the manual and found the instructions. It seems that I should go to the Python subdirectory and run python setup.py install. 

Now the program was able to find pywarpx, but amrex was not found. Do I need to run setup.py for amrex as well? Where is that file located?

It also raised an exception saying that 'Dimensionality '3d' was not compiled in this Python install. Please recompile with -DWarpX_DIMS=3'. 

I compiled WarpX with 
-DWarpX_DIMS="1;2;RZ;3" -DWarpX_PSATD=ON -DWarpX_PYTHON=ON USE_PYTHON_MAIN=TRUE



[log.txt](https://github.com/ECP-WarpX/WarpX/files/14236821/log.txt)

**Comment by ax3l** [comment]
Thanks for reaching out @rl3418 ! :) 

> I looked through the manual and found the instructions. It seems that I should go to the Python subdirectory and run python setup.py install.

That sounds outdated to me, which sub-page of the manual did you follow (link)?

Here is the manual for desktop users:
https://warpx.readthedocs.io/en/latest/install/users.html#using-the-conda-package

The manual for developers:
https://warpx.readthedocs.io/en/latest/install/cmake.html#picmi-python-bindings

And the manual for HPC systems like Perlmutter:
https://warpx.readthedocs.io/en/latest/install/hpc/perlmutter.html#compilation

In short:
- if you install with a package manager, Python is set for you.
- if you compile from source, configure with `-DWarpX_PYTHON=ON` and install the Python module with `cmake --build build --target pip_install`.

If you compile from source, I recommend that you compile inside either a Python virtual environment or inside a conda environment, whatever fits your workflow better. Without a virtual environment of some kind, your Python packages will become a bit uncontrollable/messy quickly and CMake might pick up a random Python from your system (unless you pass it with `-DPython_EXECUTABLE=$(which python3)`).

**Comment by rl3418** [comment]
https://warpx.readthedocs.io/en/latest/developers/gnumake/python.html#for-a-full-python-installation-of-warpx

This is the instructions I followed. I ran python setup.py install under the WarpX/Python directory. To compile WarpX I was running cmake --build build -j 4. I'll try cmake --build build --target pip_install then.

**Comment by rl3418** [comment]
I tried with --target pip_install. I'm able to import pywarpx, but it still doesn't recognize the amrex module.

**Comment by rl3418** [comment]
I managed to get warpx working with python using cmake --build build --target pip_install.
